### ISSUE TITLE
proto: UnmarshalText cannot unmarshal into Go map

### ISSUE THREAD

**Issue by EricLagergren** [gogo/protobuf#200]
I've a custom type that's a map. When UnmarshalText is called it calls readStruct which calls the readAny method around line 637. This uses props.mkeyprop which in other circumstances would be a pointer to the property of a map key, but since it's a Go map not a protobuf map there's no property and a panic occurs.

**Comment by awalterschulze** [comment]
Do you want to attempt a fix?

**Comment by EricLagergren** [comment]
Yes and no. I was about to yesterday but I had a bunch of other bugs to fix... and I'm not sure when, exactly, I'll be able to get around to it. (I might be able to _outside_ of work.)

**Comment by awalterschulze** [comment]
Cool well if you get time that will be great. In the mean time I have added  these two issues to the documentation for customtype.

**Comment by awalterschulze** [comment]
See here the camelcased function is called
https://github.com/gogo/letmegrpc/blob/master/html/html.go#L137

**Comment by awalterschulze** [comment]
Here might be where you would like to change the message name to also call the camelcase function
https://github.com/gogo/letmegrpc/blob/master/html/html.go#L173
https://github.com/gogo/letmegrpc/blob/master/html/html.go#L185
https://github.com/gogo/letmegrpc/blob/master/html/html.go#L190
etc

**Comment by awalterschulze** [comment]
Is it possible to provide a test?
